### PR TITLE
Nix: remove duplicate options from modules

### LINKED ISSUE #4004: remove home-manager module
### Description

We should remove the home-manager module from this repo. The hyprland module that exists in home-manager is strictly better than the one here, and having two different modules leads to some confusion (~~see https://github.com/nix-community/home-manager/issues/4716~~ that wasn't the user's problem but I think the point still stands). The migration path is also easy since users of the module from this repo can just keep shoving everything into `extraConfig`.

cc @fufexan

### PR DESCRIPTION
**Pull request by fufexan**
#### Describe your PR, what does it fix/add?

Removes the duplicated (and potentially out of date) code for the NixOS and Home
Manager modules.

The modules will still continue to work, with no breaking changes. All they do
for now is set the flake packages in the modules (for `hyprland` and `xdph`).

Fixes #4004.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

Code is untested, so please do so if you have the time.

#### Is it ready for merging, or does it need work?

Not until tested. Further discussion is encouraged.

### CODE REVIEW

**Comment by NotAShelf** [comment]
from what I understand, this makes Hyprland's nixos and home-manager modules follow downstream modules

wouldn't it make more sense to develop in this repo first, then push downstream as necessary?

**Comment by fufexan** [comment]
Options we add here will eventually make their way into the upstream modules, so this will be a "staging" place where new options are added, tested and rolled out to upstream.

Are you suggesting we update the current modules to reflect the upstream ones (since the only difference right now is the removal of `enableNvidiaPatches`, which will get into upstream in the next release)?

**Comment by andresilva** [comment]
Haven't tested this yet so not reviewing, but IMO this is the way to go. I don't see the point of duplicating code just for the sake of it. If our intention is to contribute changes done here to upstream (and I think that's the case since the arguments were that upstream is slow), then we should only duplicate code to the extent that it is needed. We can always copy back code from upstream when needed, in the meantime, just keep it simple.

**Review by andresilva — APPROVED** [review]
I tested the hm-module only and it worked as expected.

File: nix/module.nix [hunk 1]
```diff
@@ -2,96 +2,17 @@ inputs: {
...
-  options,
   ...
-}:
-with lib; let
-  cfg = config.programs.hyprland;
+}: let
   inherit (pkgs.stdenv.hostPlatform) system;
 
-  finalPortalPackage = cfg.portalPackage.override {
-    hyprland = cfg.finalPackage;
```
[the remark this answers is not in the document]
Comment (fufexan) [reply nix/module.nix:13]: Given the `xdph` package is already built with the Hyprland version in this repo, it's not needed.

[the remark this answers is not in the document]
Comment (spikespaz) [reply nix/module.nix:13]: What if `config.wayland.windowManager.hyprland.package` is overridden, and xdph uses the one from this repo? Would it not build twice? The users' overrides aren't going to make it back into the package set that xdph is built with. They would have to use an overlay to use the same NAR everywhere.

[the remark this answers is not in the document]
Comment (fufexan) [reply nix/module.nix:13]: Ah, right. We can override using `cfg.finalPackage` then.

**Comment by fufexan** [comment]
I propose merging this, and after that merging https://github.com/hyprwm/hyprland-wiki/pull/273, which I've updated to the current state of the modules and also added a link to `hyprland-nix`.

Any objections?

**Pull request merged**